bc_obps/service/tests/test_document_service_v2.py

```diff
@@ -0,0 +1,103 @@
+from service.data_access_service.document_service_v2 import DocumentDataAccessServiceV2
+from registration.utils import data_url_to_file
+from registration.models.document import Document
+from registration.models.operation import Operation
+from registration.tests.constants import MOCK_DATA_URL, MOCK_DATA_URL_2
+from service.document_service_v2 import DocumentServiceV2
+import pytest
+from service.document_service import DocumentService
+
+from model_bakery import baker
+
+pytestmark = pytest.mark.django_db
+
+
+class TestDocumentServiceV2:
+    @staticmethod
+    def test_create_operation_document():
+        # the value received by the service is a File (transformed into this in the django ninja schema)
+        file = data_url_to_file(MOCK_DATA_URL)
+        approved_user_operator = baker.make_recipe('utils.approved_user_operator')
+        operation = baker.make_recipe('utils.operation', operator=approved_user_operator.operator)
+        document, created = DocumentServiceV2.create_or_replace_operation_document(
+            approved_user_operator.user_id, operation.id, file, 'boundary_map'
+        )
+
+        assert Document.objects.count() == 1
+        assert document.type.name == 'boundary_map'
+        assert created is True
+
+    @staticmethod
+    def test_do_not_update_duplicate_operation_document():
+        approved_user_operator = baker.make_recipe('utils.approved_user_operator')
+        operation = baker.make_recipe('utils.operation', operator=approved_user_operator.operator)
+        DocumentDataAccessServiceV2.create_document(
+            approved_user_operator.user_id, data_url_to_file(MOCK_DATA_URL), 'boundary_map', operation.id
+        )
+        created_at = operation.documents.first().created_at
+
+        updated_file = data_url_to_file(MOCK_DATA_URL)
+        document, created = DocumentServiceV2.create_or_replace_operation_document(
+            approved_user_operator.user_id, operation.id, updated_file, 'boundary_map'
+        )
+
+        assert Document.objects.count() == 1
+        assert document.type.name == 'boundary_map'
+        # MOCK_DATA_URL's filename is mock.pdf. When adding files to django, the name is appended, so we just check that 'mock' in the name
+        assert document.file.name.find("mock") != -1
+        assert document.created_at == created_at
+        assert created is False
+
+    @staticmethod
+    def test_update_operation_document():
```

Comment:
could I request another unit test here to confirm that an industry_user is NOT allowed to update/view a document associated with an operation outside of their operator? (this relates to my previous comment above)